web/src/navbar_menus.js

```diff
@@ -4,54 +4,101 @@ import {page_params} from "./page_params";
 import * as personal_menu_popover from "./personal_menu_popover";
 import * as popover_menus from "./popover_menus";
 
+// Available menu options based on user type
+const popover_menus_options = page_params.is_spectator
+    ? ["signup_button", "login_button", "navbar_help_menu", "gear_menu"]
+    : ["navbar_help_menu", "gear_menu", "personal_menu"];
+
 export function is_navbar_menus_displayed() {
     return (
-        popover_menus.is_personal_menu_popover_displayed() ||
-        popover_menus.is_gear_menu_popover_displayed() ||
-        popover_menus.is_help_menu_popover_displayed()
+        document.activeElement &&
```

Comment:
`document.activeElement` should only matter for signup/login buttons. Let's keep `is_navbar_menus_displayed` the same and add another function which checks this named `can_handle_navigation_hotkey`.

Reply:
Updated as Suggested.

Reply:
I was thinking `can_handle_navigation_hotkey` will have the `|| is_navbar_menus_displayed` inside it.